From 1e91c1d4e37c05cf95058b4b3c3f352d72886f58 Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Tue, 15 Aug 2017 17:31:38 +0800
Subject: [PATCH 7/7] Fix control path where we have str as uninitialized
 string

|
/home/ubuntu/work/oe/openembedded-core/build/tmp-musl/work/i586-oe-linux-musl/elfutils/0.164-r0/elfutils-0.164/libcpu/i386_disasm.c:
In function 'i386_disasm':
|
/home/ubuntu/work/oe/openembedded-core/build/tmp-musl/work/i586-oe-linux-musl/elfutils/0.164-r0/elfutils-0.164/libcpu/i386_disasm.c:310:5:
error: 'str' may be used uninitialized in this function
[-Werror=maybe-uninitialized]
|      memcpy (buf + bufcnt, _str, _len);           \
|      ^
|
/home/ubuntu/work/oe/openembedded-core/build/tmp-musl/work/i586-oe-linux-musl/elfutils/0.164-r0/elfutils-0.164/libcpu/i386_disasm.c:709:17:
note: 'str' was declared here
|      const char *str;

Signed-off-by: Khem Raj <raj.khem@gmail.com>
Upstream-Status: Pending

Rebase to 0.170

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 libcpu/i386_disasm.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libcpu/i386_disasm.c b/libcpu/i386_disasm.c
index 831afbe..60fd6d4 100644
--- a/libcpu/i386_disasm.c
+++ b/libcpu/i386_disasm.c
@@ -821,6 +821,7 @@ i386_disasm (Ebl *ebl __attribute__((unused)),
 			    }
 			  /* Fallthrough */
 			default:
+			  str = "";
 			  assert (! "INVALID not handled");
 			}
 		    }
-- 
1.8.3.1

